### ISSUE TITLE
cannot stat ‘libopenblas_sandybridgep-r0.2.18.so’ when installing pbalign

### ISSUE THREAD

**Issue by pawelosipowski** [PacificBiosciences/pitchfork#292]
I get this with the newly cloned pitchfork: 
```
make[2]: Entering directory `/storage3/long_read_assembly/soft/pitchfork/workspace/OpenBLAS-0.2.18'
make -j 24 -f Makefile.install install
make[3]: Entering directory `/storage3/long_read_assembly/soft/pitchfork/workspace/OpenBLAS-0.2.18'
Generating openblas_config.h in /storage3/long_read_assembly/soft/pitchfork/staging/OpenBLAS-0.2.18/include
Generating f77blas.h in /storage3/long_read_assembly/soft/pitchfork/staging/OpenBLAS-0.2.18/include
Generating cblas.h in /storage3/long_read_assembly/soft/pitchfork/staging/OpenBLAS-0.2.18/include
Copying LAPACKE header files to /storage3/long_read_assembly/soft/pitchfork/staging/OpenBLAS-0.2.18/include
Copying the static library to /storage3/long_read_assembly/soft/pitchfork/staging/OpenBLAS-0.2.18/lib
Copying the shared library to /storage3/long_read_assembly/soft/pitchfork/staging/OpenBLAS-0.2.18/lib
install: cannot stat ‘libopenblas_sandybridgep-r0.2.18.so’: No such file or directory
make[3]: *** [install] Error 1
make[3]: Leaving directory `/storage3/long_read_assembly/soft/pitchfork/workspace/OpenBLAS-0.2.18'
make[2]: *** [install] Error 2
make[2]: Leaving directory `/storage3/long_read_assembly/soft/pitchfork/workspace/OpenBLAS-0.2.18'
make[1]: *** [/storage3/long_read_assembly/soft/pitchfork/deployment/var/pkg/OpenBLAS-0.2.18] Error 1
make[1]: Leaving directory `/storage3/long_read_assembly/soft/pitchfork/ports/thirdparty/openblas'
make: *** [openblas] Error 2
```

**Comment by mhsieh** [comment]
Hi. Which distribution/version of linux are you using?

**Comment by pawelosipowski** [comment]
Hi, I'm using Ubuntu 14.04

**Comment by mhsieh** [comment]
this looks like that this might be some latency issue in your file system. Do you mind trying to use less concurrent make thread? for example start from `make -j1` or `make -j2`.

**Comment by pawelosipowski** [comment]
the same effect after I tried

**Comment by mhsieh** [comment]
You probably need to  `make distclean-openblas` and do the make again.

**Comment by pawelosipowski** [comment]
I did and no success again
```
test -e ports/pacbio/openblas     && make -C ports/pacbio/openblas     do-distclean || true
test -e ports/thirdparty/openblas && make -C ports/thirdparty/openblas do-distclean || true
make[1]: Entering directory `/storage3/long_read_assembly/soft/pitchfork/ports/thirdparty/openblas'
make[1]: *** No rule to make target `do-distclean'.  Stop.
make[1]: Leaving directory `/storage3/long_read_assembly/soft/pitchfork/ports/thirdparty/openblas'
test -e ports/python/openblas     && make -C ports/python/openblas     do-distclean || true
```

**Comment by mhsieh** [comment]
not to worry, I forgot to commit the distclean back. It will be available in a minute.

**Comment by mhsieh** [comment]
In this case, you can update the pitchfork by doing `git pull --rebase`

**Comment by pawelosipowski** [comment]
I fallowed but again
```
Copying the shared library to /storage3/long_read_assembly/soft/pitchfork/staging/OpenBLAS-0.2.18/lib
install: cannot stat ‘libopenblas_sandybridgep-r0.2.18.so’: No such file or directory
make[3]: *** [install] Error 1
make[3]: Leaving directory `/storage3/long_read_assembly/soft/pitchfork/workspace/OpenBLAS-0.2.18'
make[2]: *** [install] Error 2
make[2]: Leaving directory `/storage3/long_read_assembly/soft/pitchfork/workspace/OpenBLAS-0.2.18'
make[1]: *** [/storage3/long_read_assembly/soft/pitchfork/deployment/var/pkg/OpenBLAS-0.2.18] Error 1
make[1]: Leaving directory `/storage3/long_read_assembly/soft/pitchfork/ports/thirdparty/openblas'
make: *** [openblas] Error 2
```

**Comment by mhsieh** [comment]
Try doing _startover and then continue the build, I don't believe that this is a clean build since the building openblas part should take you more than 12 minutes to do it.

**Comment by mhsieh** [comment]
```
mhsieh@localhost14:~/pitchfork$ lsb_release -d
Description:    Ubuntu 14.04 LTS 
mhsieh@localhost14:~/pitchfork$ make -j24 openblas
make -f mk/init.mk init
make[1]: Entering directory `/scratch/mhsieh/pitchfork'
/scratch/mhsieh/pitchfork/bin/checkSystem
/scratch/mhsieh/pitchfork/bin/checkCC gcc
mkdir -p "/scratch/mhsieh/pfworkspace"
mkdir -p "/scratch/mhsieh/distfiles/pitchfork"
mkdir -p "/scratch/mhsieh/pitchfork/staging"
mkdir -p "/scratch/mhsieh/pfbuild_FC/bin"
mkdir -p "/scratch/mhsieh/pfbuild_FC/etc"
mkdir -p "/scratch/mhsieh/pfbuild_FC/include"
mkdir -p "/scratch/mhsieh/pfbuild_FC/lib"
mkdir -p "/scratch/mhsieh/pfbuild_FC/lib/pkgconfig"
mkdir -p "/scratch/mhsieh/pfbuild_FC/share"
mkdir -p "/scratch/mhsieh/pfbuild_FC/var/pkg"
echo "export   LD_LIBRARY_PATH=/scratch/mhsieh/pfbuild_FC/lib:\$LD_LIBRARY_PATH"  |sed -e 's/::*/:/g' > "/scratch/mhsieh/pfbuild_FC/setup-env.sh"
echo "export PATH=/scratch/mhsieh/pfbuild_FC/bin:\$PATH"|sed -e 's/::*/:/g' >> "/scratch/mhsieh/pfbuild_FC/setup-env.sh"
echo "unset PYTHONPATH"
unset PYTHONPATH
make[1]: Leaving directory `/scratch/mhsieh/pitchfork'
touch mk//../initialized.o
make -C ports/thirdparty/ccache do-install
make[1]: Entering directory `/scratch/mhsieh/pitchfork/ports/thirdparty/ccache'
md5sum -c MD5SUM || exit
ccache-3.2.4.tar.gz: OK
tar zxf ccache-3.2.4.tar.gz -C /scratch/mhsieh/pfworkspace
(cd /scratch/mhsieh/pfworkspace/ccache-3.2.4 && ./configure --prefix=/scratch/mhsieh/pfbuild_FC --with-bundled-zlib) >& config.log || /scratch/mhsieh/pitchfork/bin/diewith config.log
make -C /scratch/mhsieh/pfworkspace/ccache-3.2.4 >& build.log || /scratch/mhsieh/pitchfork/bin/diewith build.log 
rm -rf /scratch/mhsieh/pitchfork/staging/ccache-3.2.4
make -C /scratch/mhsieh/pfworkspace/ccache-3.2.4 prefix=/scratch/mhsieh/pitchfork/staging/ccache-3.2.4 install >& install.log || /scratch/mhsieh/pitchfork/bin/diewith install.log
cd /scratch/mhsieh/pitchfork/staging/ccache-3.2.4/bin && ln -sfn ccache `basename gcc` 
cd /scratch/mhsieh/pitchfork/staging/ccache-3.2.4/bin && ln -sfn ccache `basename g++`
cd /scratch/mhsieh/pitchfork/staging/ccache-3.2.4/bin && ln -sfn ccache `basename gfortran`
cd /scratch/mhsieh/pitchfork/staging/ccache-3.2.4/bin && ln -sfn ccache cpp
cd /scratch/mhsieh/pitchfork/staging/ccache-3.2.4/bin && ln -sfn ccache CC
cd /scratch/mhsieh/pitchfork/staging/ccache-3.2.4 && tar cf - * | tar xf - -C /scratch/mhsieh/pfbuild_FC
find /scratch/mhsieh/pitchfork/staging/ccache-3.2.4 ! -type d|awk -F '/scratch/mhsieh/pitchfork/staging/ccache-3.2.4/' '{print $2}' > /scratch/mhsieh/pfbuild_FC/var/pkg/ccache-3.2.4
make[1]: Leaving directory `/scratch/mhsieh/pitchfork/ports/thirdparty/ccache' 
make -C ports/thirdparty/openblas do-install
make[1]: Entering directory `/scratch/mhsieh/pitchfork/ports/thirdparty/openblas'
/scratch/mhsieh/pitchfork/bin/pitchfork fetch --url https://github.com/xianyi/OpenBLAS/archive/v0.2.18.tar.gz
using local v0.2.18.tar.gz
md5sum -c MD5SUM || exit
v0.2.18.tar.gz: OK
tar zxf v0.2.18.tar.gz -C /scratch/mhsieh/pfworkspace
make -C /scratch/mhsieh/pfworkspace/OpenBLAS-0.2.18 MAKE_NB_JOBS=2 2>&1 | tee build.log | /scratch/mhsieh/pitchfork/bin/activity || /scratch/mhsieh/pitchfork/bin/diewith build.log
rm -rf /scratch/mhsieh/pitchfork/staging/OpenBLAS-0.2.18 
make -j1 -C /scratch/mhsieh/pfworkspace/OpenBLAS-0.2.18 PREFIX=/scratch/mhsieh/pitchfork/staging/OpenBLAS-0.2.18 install >& install.log || /scratch/mhsieh/pitchfork/bin/diewith install.log   
cd /scratch/mhsieh/pitchfork/staging/OpenBLAS-0.2.18 && tar cf - * | tar xf - -C /scratch/mhsieh/pfbuild_FC
find /scratch/mhsieh/pitchfork/staging/OpenBLAS-0.2.18 ! -type d|awk -F '/scratch/mhsieh/pitchfork/staging/OpenBLAS-0.2.18/' '{print $2}' > /scratch/mhsieh/pfbuild_FC/var/pkg/OpenBLAS-0.2.18
make[1]: Leaving directory `/scratch/mhsieh/pitchfork/ports/thirdparty/openblas'
```

**Comment by pawelosipowski** [comment]
I did this and it's seems to work. Great!

**Comment by mhsieh** [comment]
let's what do you have in /storage3/long_read_assembly/soft/pitchfork/workspace/OpenBLAS-0.2.18 shall we?

```
ls -l /storage3/long_read_assembly/soft/pitchfork/workspace/OpenBLAS-0.2.18/libopenblas.*
```

**Comment by pawelosipowski** [comment]
sure 
```
$ ls -l /storage3/long_read_assembly/soft/pitchfork/workspace/OpenBLAS-0.2.18/libopenblas.*
lrwxrwxrwx 1 pawel pawel 34 Sep 24 10:34 /storage3/long_read_assembly/soft/pitchfork/workspace/OpenBLAS-0.2.18/libopenblas.a -> libopenblas_sandybridgep-r0.2.18.a
```

**Comment by mhsieh** [comment]
This might be a problem from openblas side where I don't have enough bandwidth to spend time on. Can you just download it from this https://dl.dropboxusercontent.com/u/174258/openblas.tar and see if that un-blocks you.
```
$ wget https://dl.dropboxusercontent.com/u/174258/openblas.tar
$ tar xf openblas.tar -C /storage3/long_read_assembly/soft/pitchfork/deployment
```

**Comment by mhsieh** [comment]
Let me know if that works or not. Generally speaking, port software for multiple use cases is hard. I don't have the hardware you currently build upon, and openblas is supposed to be very sensitive to hardware configuration.

**Comment by pawelosipowski** [comment]
It worked fine. Thanks

**Comment by mhsieh** [comment]
Feel free to close this issue, I will close this after I am able to find a similar hardware to reproduce.

**Comment by mchaisso** [comment]
Note I had the same problem, and resolved this by running make in the OpenBLAS-0.2.18 workspace directory, which was able to build the .so file.  Not sure why it worked manually rather than from root pitchfork dir.
